### PR TITLE
MAINTENANCE: Improve build scripts

### PR DESCRIPTION
**Pull request by ThomasSchwengler**
* Upgrade gradle to `5.6.4`
* Upgrade android gradle plugin to `3.5.2`
* Upgrade support library version to `28.0.0`
* Upgrade LibGDX to `1.9.10`
  * This should fix some warnings with calls to graylisted functions
* Upgrade checkstyle to `8.26`
* Upgrade kotlin gradle plugin to `1.3.60`
* Move `fabric.properties` to a template file and generate it from buildscript
* Fix some kotlin warnings
* Remove obsolete property from `gradle.properties`
* Upgrade fabric gradle plugin to fix some warnings
* Upgrade `detekt` to a stable version (was RC before)
  * Update detekt config from default config, migrate `test-pattern` excludes
* Move androidemulators specific configuration into a separate file
* Group ext variables into one block
* Move `detekt` configuration into `code_quality_tasks`
* Create a `copyGoogleServicesFile` function to improve readability
* Simplify flavor declaration
  * Group `buildConfigField` properties
  * Move `buildConfigField` properties into the default configuration and overwrite in flavors
  * Remove unused `FEATURE_CLOUD_MESSAGING_ENABLED` build config field
* Add dependency to `copyAndroidNatives` as done in the libGDX example gradle file

### CODE REVIEW

**Comment by PlantinumX** [comment]
>   * Move `buildConfigField` properties into the default configuration and overwrite in flavors
Does this work? I tried it several times in ticket CATROID-329, it did not work.

**Comment by ThomasSchwengler** [comment]
@PlantinumX It works as far as I have seen
image

File: catroid/config/detekt.yml [hunk 1]
```diff
@@ -1,29 +1,5 @@
 autoCorrect: true
 
-test-pattern: # Configure exclusions for test sources
```
Comment (ThomasSchwengler) [inline catroid/config/detekt.yml]: Test pattern has been migrated according to this guideline:
https://arturbosch.github.io/detekt/howto-migratetestpattern.html

**Comment by wslany** [comment]
@ThomasSchwengler Should removing the conflicts make this mergeable?

**Comment by thmq** [comment]
@ThomasSchwengler can I haz rebase, plz?

**Comment by ThomasSchwengler** [comment]
> @ThomasSchwengler can I haz rebase, plz?

Done :)

**Comment by wslany** [comment]
Thanks!

**Pull request merged**